### ISSUE TITLE
Replace integrated tor connectivity with orbot requirement?

### ISSUE THREAD

**Issue by ardevd** [stakwork/sphinx-kotlin#379]
The iOS and Android versions of Sphinx come bundled with their own Tor connectivity mechanisms. For iOS it makes sense but wouldn't it make more sense to just rely on the user using Orbot for the Sphinx app on Android? You can remove a bunch of code and arguably make the end user's life easier in the process?

**Comment by gonzaloaune** [comment]
Hey @ardevd, we found many issues in the past using Orbot, I personally had to restart not only the app but even the phone sometimes and I couldn't get a stable connection using Orbot. I know many of our users had the same problem and ultimately led to them giving up on trying our app.

I know it makes sense from one point of view to use Orbot but then the app will have a dependency on another app, what happens if Orbot ceases to exist? I think having the Tor library integrated in the app directly makes a lot of sense but let me know if I'm missing something!

**Comment by ardevd** [comment]
I agree with your argument. However, those of us who rely on tor based services tend to use orbot extensively anyway, making an integrated tor solution in Sphinx redundant. 

However, how about making it possible to disable integrated tor? Those of us using Orbot would be happy and those who don't can continue to use the integrated one. This obviously invalidates my point of removing code and instead requires adding some, but I still think it could be valuable. 

For example, I already need to run BlueWallet with orbot.

**Comment by gonzaloaune** [comment]
That sounds like a good compromise, if you don't mind me asking, why would you want to run Orbot instead of using the integrated solution? is this for specific configuration that can be done in Orbot? I haven't used it extensively... 

If that's the case, I think adding the option to disable the integrated tor makes sense but I wouldn't remove it altogether, some people will be just as happy to run Sphinx with the integrated tor solution and won't bother using Orbot

**Comment by ardevd** [comment]
Using orbot just seems more sensible when I already use it for other apps. Not sure what sort of battery drain the associated tor-service comes with, but I prefer having a one-stop tor-shop through orbot where the VPN mode setup lets me easily toggle which apps should be routed through tor.

I agree that letting the user toggle integrated tor on/off makes more sense than removing the integrated tor service as I originally proposed.
